### PR TITLE
Fix issue with Elemental analysis not supporting legacy data

### LINKED ISSUE #27430: Elemental analysis does not support legacy data
**Original reporter Adrian Hillier** 

The elemental analysis GUI has difficulty dealing with legacy data. For instance, if the data set is missing the Delayed data, an error is raised, which will prevent any data from being plotted. This issue was encountered by Adrian Hillier while processing some old data. This issue is due to the hard-coding of values into the elemental analysis loading method. 

### Steps to reproduce the behavior
Make of a copy of some elemental analysis data (this is not readily available)
Delete the delayed response (files ending in 20)
Open up the elemental analysis GUI
Add the modified run data to the user directories
Load the modified run (the one where you've deleted the delayed response)
The GUI will not be able to handle this, and an error will be raised
### Platforms affected
Windows and Linux (Ubuntu), and presumably OSX

### PR DESCRIPTION
**Pull request by StephenSmith25**
The elemental analysis GUI had difficulty dealing with legacy data. For instance, if the data set was missing the delayed response, an error was raised, which prevented any data from being plotted. This issue was encountered by Adrian Hillier while processing some old data. 

This PR fixes this issue, by associating a workspace (of Y(x)=0) to the missing data. This enables  elemental analysis to continue as expected.

**Report to:** Adrian Hillier 

**To test:**
Make of a copy of some elemental analysis data (this is not readily available)
Delete the delayed response (files ending in 20)
Open up the elemental analysis GUI
Add the modified run data to the user directories
Load the modified run (the one where you've deleted the delayed response)
The GUI should be able to handle this. Check that all the plots work. The delayed response should be plotted with (Y=0)

Fixes #27430.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Matthew-Andrew — DISMISSED** [review]
Looks good to go :+1:

**Comment by AnthonyLim23** [comment]
I think we should add a test to check it can load legacy data

**Comment by AnthonyLim23** [comment]
got a conflict

**Review by AnthonyLim23 — APPROVED** [review]

File: scripts/Muon/GUI/ElementalAnalysis/LoadWidget/load_utils.py [hunk 1]
```diff
@@ -117,43 +125,45 @@ def merge_workspaces(run, workspaces):
...
-            ws = mantid.mtd[workspace_list[i]]
-            max_num_bins = max(ws.blocksize(), max_num_bins)
+        for ws_name in workspace_list:
+            if ws_name:
+                ws = mantid.mtd[ws_name]
+                max_num_bins = max(ws.blocksize(), max_num_bins)
 
         # create single ws for the merged data, use original ws as a template
-        merged_ws = mantid.WorkspaceFactory.create(mantid.mtd[workspace_list[0]], NVectors=num_workspaces,
+        template_ws = next(ws for ws in workspace_list if ws is not None)
```
Comment (DanNixon) [inline scripts/Muon/GUI/ElementalAnalysis/LoadWidget/load_utils.py]: :+1: I hadn't considered doing this when you want to use a generator as a filter.

**Pull request merged**